### DISCUSSION TITLE
Bug report: Can't upgrade after exceeding limits, stuck in infinite restore loop

### DISCUSSION THREAD

**Discussion by X4** [supabase/supabase#13851] [General]
**CRITICAL**
We ran into this issue and need a dump asap. Ticket created already, waiting for an answer. Here to describe how it happened and to fix the issue at it's root. It may affect many more.

**Steps to reproduce:**

1. Exceeed free limits
2. Remove read-only mode, to remove data
3. Exceed free io limits, whilst trying to delete data
4. Database get's paused by system
5. Infinite loop of Restoring the database 
6. Upgrade impossible, Backups and many other services impossible to reach either

**Comment by cedricmay** [comment]
+1 Same here, we can't access our database anymore as well.
